File: Delete.java
public class Delete {
    void deleteNode(Node del) {
        if (del == null) {
            return;
        }

        while (del.next != null && del.next.next != null) {
            del.data = del.next.data;
            del = del.next;
        }
        del.data = del.next.data;
        del.next = null;

    }
}

File: delete.py
#Time Complexity : O(1)
class Solution:
    #Function to delete a node without any reference to head pointer.
    def deleteNode(self,curr_node):
        nextNode = curr_node.next
        curr_node.data = nextNode.data
        curr_node.next = curr_node.next.next
        
